src/repositories/genre_repository.go

```diff
@@ -0,0 +1,207 @@
+	DatabaseManager managers.DatabaseManagerI
+}
+
+// Genre
+func (mr *GenreRepository) GetGenres() (*[]model.Genres, *models.KTSError) {
+	var genres []model.Genres
+
+	// Create the query
+	stmt := mysql.SELECT(
+		table.Genres.AllColumns,
+	).FROM(
+		table.Genres,
+	)
+
+	// Execute the query
+	err := stmt.Query(mr.DatabaseManager.GetDatabaseConnection(), &genres)
+	if err != nil {
+		return nil, kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if len(genres) == 0 {
+		return nil, kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return &genres, nil
+}
+
+func (mr *GenreRepository) GetGenreByName(name *string) (*model.Genres, *models.KTSError) {
+	var genre model.Genres
+
+	// Create the query
+	stmt := mysql.SELECT(
+		table.Genres.AllColumns,
+	).FROM(
+		table.Genres,
+	).WHERE(
+		table.Genres.GenreName.EQ(utils.MySqlString(*name)),
+	)
+
+	// Execute the query
+	err := stmt.Query(mr.DatabaseManager.GetDatabaseConnection(), &genre)
+	if err != nil {
+		if err.Error() == "qrm: no rows in result set" {
+			return nil, kts_errors.KTS_MOVIE_NOT_FOUND
+		}
+		return nil, kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	return &genre, nil
+}
+
+func (mr *GenreRepository) CreateGenre(name *string) *models.KTSError {
+	// Create the insert statement
+	insertQuery := table.Genres.INSERT(table.Genres.GenreName).
+		VALUES(name)
+
+	// Execute the query
+	rows, err := insertQuery.Exec(mr.DatabaseManager.GetDatabaseConnection())
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	rowsAff, err := rows.RowsAffected()
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if rowsAff == 0 {
+		return kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return nil
+}
+
+func (mr *GenreRepository) UpdateGenre(genre *model.Genres) *models.KTSError {
+
+	// Create the update statement
+	updateQuery := table.Genres.UPDATE(table.Genres.GenreName).
+		SET(genre.GenreName).
+		WHERE(table.Genres.ID.EQ(utils.MysqlUuid(genre.ID)))
+
+	// Execute the query
+	rows, err := updateQuery.Exec(mr.DatabaseManager.GetDatabaseConnection())
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	rowsAff, err := rows.RowsAffected()
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if rowsAff == 0 {
+		return kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return nil
+}
+
+func (mr *GenreRepository) DeleteGenre(genreId *uuid.UUID) *models.KTSError {
+	
+
+	// Create the delete statement
+	deleteQuery := table.Genres.DELETE().
+	WHERE(table.Genres.ID.EQ(utils.MysqlUuid(genreId)))
+
+	// Execute the query
+	rows, err := deleteQuery.Exec(mr.DatabaseManager.GetDatabaseConnection())
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	rowsAff, err := rows.RowsAffected()
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if rowsAff == 0 {
+		return kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return nil
+}
+
+// One Genre with all table.Movies
+func (mr *GenreRepository) GetGenreByNameWithMovies(genreName *string) (*models.GenreWithMovies, *models.KTSError) {
```

Comment:
For waht could this be needed, personally wouldnt query by name, rather by id

Reply:
Filter. That way they do not have to sort and filter themself

Reply:
Aber selbst dann sollten die mit der id anfragen und nicht mit dem namen

Reply:
Aber woher wissen die die UUID? Die müssten dann erst mit GetGenreByName die UUID holen und dann die Filme dazu suchen. 
Mit der Methode spart man sich request. Außerdem haben die genau so eine Filterung mit dem Namen

Reply:
lass uns da später nochmal reden

Reply:
keeping this method for now